monai/transforms/post/array.py

```diff
@@ -936,3 +938,33 @@ def __call__(self, image: NdarrayOrTensor) -> torch.Tensor:
         grads = convert_to_dst_type(grads.squeeze(0), image_tensor)[0]
 
         return grads
+
+
+class DistanceTransformEDT(Transform):
+    """
+    Applies the Euclidean distance transform on the input.
+
+    Either GPU based with CuPy / cuCIM or CPU based with scipy.ndimage.
+    Choice only depends on cuCIM being available.
+    Note that the calculations can deviate, for details look into the cuCIM about distance_transform_edt().
+    """
+
+    backend = [TransformBackends.NUMPY, TransformBackends.CUPY]
+
+    def __init__(self, sampling: None | float | list[float] = None, force_scipy: bool = False) -> None:
+        super().__init__()
+        self.force_scipy = force_scipy
+        self.sampling = sampling
+
+    def __call__(self, img: NdarrayOrTensor) -> NdarrayOrTensor:
+        """
+        Args:
+            img: Input image on which the distance transform shall be run.
+                For CuPy shape must be ([channel, ]spatial_dim1, spatial_dim2 [, spatial_dim3]), so 2D, 3D and 4D input only.
```

Comment:
We should assume the img here has a channel-first shape just like most MONAI transforms do.